salt/modules/http.py

```diff
@@ -79,6 +79,10 @@ def wait_for_successful_query(url, wait_for=300, **kwargs):
                 raise caught_exception  # pylint: disable=E0702
 
             return result
+        else:
```

Comment:
Why not just put an elif here instead of else and then an if inside it?

Reply:
I just copy-pasted this section from the state file. Keeping them the same sounds reasonable to me OR we could change them both...